cdap-common/bin/functions.sh

```diff
@@ -1020,7 +1020,12 @@ cdap_sdk_start() {
     CDAP_SDK_DEFAULT_JVM_OPTS="-Xmx2048m"
   fi
 
-  eval split_jvm_opts ${CDAP_SDK_DEFAULT_JVM_OPTS} ${CDAP_SDK_OPTS} ${JAVA_OPTS}
+  SDK_GC_OPTS="-verbose:gc -Xloggc:${LOG_DIR}/gc.log -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=10 -XX:GCLogFileSize=1M"
+  if [[ ${HEAPDUMP_ON_OOM} == true ]]; then
```

Comment:
When will this be not true?

Reply:
User can override it:
https://github.com/caskdata/cdap/blob/develop/cdap-distributions/src/etc/cdap/conf.dist/cdap-env.sh#L41-L43

Reply:
We implemented that ability to disable it as a response to this comment:
https://github.com/caskdata/cdap/pull/9196#issuecomment-313479613

Reply:
BTW, do we kill the process on OOM as well?

Reply:
Currently, no.

Reply:
Created a JIRA for it: https://issues.cask.co/browse/CDAP-14097